db/memory/src/main/java/org/commonjava/indy/mem/data/MemoryStoreDataManager.java

```diff
@@ -354,11 +357,69 @@ public void clear( final ChangeSummary summary )
     @Override
     public RemoteRepository findRemoteRepository( final String url )
     {
+
         List<Map.Entry<StoreKey, ArtifactStore>> copy = new ArrayList<>( stores.entrySet() );
 
+        /*
+           This filter do these things:
+             * First compare ip, if ip same, and the path(without last slash) same too, the repo is found
+             * If ip not same, then compare the url without scheme and last slash (if has) to find the repo
+         */
+        Predicate<Map.Entry<StoreKey, ArtifactStore>> findingRepoFilter = e -> {
+            if ( ( remote == e.getValue().getKey().getType() ) )
+            {
+                final UrlInfo urlInfo = new UrlInfo( url );
+                final String targetUrl = ( (RemoteRepository) e.getValue() ).getUrl();
+                final UrlInfo targetUrlInfo = new UrlInfo( targetUrl );
+                final String ipForUrl;
+                final String ipForTargetUrl;
+                try
+                {
+                    ipForUrl = urlInfo.getIpForUrl();
+                    ipForTargetUrl = targetUrlInfo.getIpForUrl();
+                }
+                catch ( UnknownHostException ue )
```

Comment:
I think if we get UnknownHostException, we should just fall back to straight string URL analysis and forget the IP comparison. Worst case is we get multiple repositories pointing at the same thing from different DNS CNAMEs...but if we return false here we might end up with multiple repositories pointing at the exact same URL, just because the DNS server was down (or the network was)